common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -281,14 +285,26 @@ def _gated_content_staff_banner(self):
             if content_milestones and self.runtime.user_is_staff:
                 return banner_text
 
+    def _get_course_end(self):
+        """
+        Helper to get course end, because I can't figure out an easier way to do it. Open to suggestions.
+        """
+        current = self
+        while current.location.block_type != 'course':
+            current = current.get_parent()
+            if current == None:
+                raise Exception("Key assumption violated - went past root of course tree without hitting course block")
+        return current.end
```

Comment:
If anyone knows of a better way to get the current block's course's end date, I'm all ears. Otherwise we'll need to clean up the wording and type of the exception